### ISSUE TITLE
set() - setting columns to null not working

### ISSUE THREAD

**Issue by martinbel** [Rdatatable/data.table#1221]
I couldn't set to null a few columns. I normally do this in this way, so I think the code is ok. 

Here is a head of the data:
https://gist.githubusercontent.com/martinbel/a6adfd1451fb0283b28a/raw/15b37de043ef0c9d186d6312f03c701288bab0b0/issue_data.R

Hope this helps.

> class(X_all)
[1] "data.table" "data.frame"

> drop_cols = c('campaign_details', 'product_description', 'client_id')
> set(X_all, i=NULL, j=drop_cols, value=NULL)
Error in set(X_all, i = NULL, j = drop_cols, value = NULL) : 
  Internal error, please report (including result of sessionInfo()) to datatable-help: oldtncol (0) < oldncol (58) but tl of class is marked.
In addition: Warning message:
In set(X_all, i = NULL, j = drop_cols, value = NULL) :
  Adding new column 'client_id' then assigning NULL (deleting it).
> sessionInfo()
R version 3.2.0 (2015-04-16)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: OS X 10.9.5 (Mavericks)

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] caret_6.0-47     lattice_0.20-31  glmnet_2.0-2     foreach_1.4.2    Matrix_1.2-0    
 [6] scales_0.2.5     ggplot2_1.0.1    data.table_1.9.4 reshape2_1.4.1   plyr_1.8.3      

loaded via a namespace (and not attached):
 [1] Rcpp_0.11.6         magrittr_1.5        splines_3.2.0       MASS_7.3-40         munsell_0.4.2      
 [6] colorspace_1.2-6    minqa_1.2.4         car_2.0-25          stringr_1.0.0       tools_3.2.0        
[11] parallel_3.2.0      pbkrtest_0.4-2      nnet_7.3-9          grid_3.2.0          gtable_0.1.2       
[16] nlme_3.1-120        mgcv_1.8-6          quantreg_5.11       gtools_3.5.0        iterators_1.0.7    
[21] lme4_1.1-7          digest_0.6.8        nloptr_1.0.4        codetools_0.2-11    stringi_0.5-5      
[26] BradleyTerry2_1.0-6 SparseM_1.6         brglm_0.5-9         chron_2.3-45        proto_0.3-10       
> grep('camp', names(X_all), v=T)
[1] "campaign"             "campaign_description" "campaign_details"

**Comment by franknarf1** [comment]
The `j` argument of `set` takes one column at a time.

**Comment by martinbel** [comment]
So the issue was the client id column wasn't in the X_all data.table. After removing this field from drop_cols character vector, it worked ok (without the error). 
Anyway. I though it could be useful to report it.

**Comment by jangorecki** [comment]
@martinbel thanks for update. It seems to be fixed in *1.9.5* version.
```r
library(data.table)
packageVersion("data.table")
# [1] ‘1.9.5’
d <- data.table(a=1,b=2)
drop_cols <- c("b","c")
set(d, i=NULL, j=drop_cols, value=NULL)
# Warning message:
# In set(d, i = NULL, j = drop_cols, value = NULL) :
#     Adding new column 'c' then assigning NULL (deleting it).
print(d)
#    a
# 1: 1
```
You can try on 1.9.5, there are several others fixes and lots of improvements.
Feel free to close issue.

**Comment by martinbel** [comment]
OK. Perfect. Thanks
